contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1645,6 +1645,12 @@ object_id(PG_FUNCTION_ARGS)
 			}
 		}
 	}
+
+	if (is_temp_object && !result)
+	{
+		result = get_relname_relid((const char *) object_name, LookupNamespaceNoError("pg_temp"));
+	}
```

Comment:
I think it is better to include this IF block in the above if..else blocks where we are already checking whether it is a temp object i.e.,  [#L1537](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2159/files#diff-14053c36507852e9e6bec9aaf6879220d4e22efea701990e9bfc5ac6289523d4R1537) and [#L1645](https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/2159/files#diff-14053c36507852e9e6bec9aaf6879220d4e22efea701990e9bfc5ac6289523d4R1605)

Reply:
+1, we should be doing this in specific blocks corresponding to temp objects to avoid unnecessary lookups.